File: pinyougou-user-web/src/main/webapp/js/controller/userController.js
 //控制层 
app.controller('userController' ,function($scope,$controller ,userService){
	
	$controller('baseController',{$scope:$scope});//继承
	
    //读取列表数据绑定到表单中  
	$scope.findAll=function(){
		userService.findAll().success(
			function(response){
				$scope.list=response;
			}			
		);
	};
	
	//分页
	$scope.findPage=function(page,rows){			
		userService.findPage(page,rows).success(
			function(response){
				$scope.list=response.rows;	
				$scope.paginationConf.totalItems=response.total;//更新总记录数
			}			
		);
	};
	
	//查询实体 
	$scope.findOne=function(id){				
		userService.findOne(id).success(
			function(response){
				$scope.entity= response;					
			}
		);				
	};
	
	/*
		进行注册的方法
	 */
	$scope.add=function(){
		//调用userService中的add方法
		userService.add($scope.entity,$scope.code).success(function (response) {
			if (response.success()){
				alert(response.message);
			} else {
				alert(response.message);
			}
		})
	};
	
	 
	//批量删除 
	$scope.dele=function(){			
		//获取选中的复选框			
		userService.dele( $scope.selectIds ).success(
			function(response){
				if(response.success){
					$scope.reloadList();//刷新列表
					$scope.selectIds=[];
				}						
			}		
		);				
	}
	
	$scope.searchEntity={};//定义搜索对象 
	
	//搜索
	$scope.search=function(page,rows){			
		userService.search(page,rows,$scope.searchEntity).success(
			function(response){
				$scope.list=response.rows;	
				$scope.paginationConf.totalItems=response.total;//更新总记录数
			}			
		);
	};

	/*
		1.判断两次输入的密码是否一致,输入一致提交信息，输入不一致打断方法
	 */
    $scope.reg=function () {
		if ($scope.entity.password != $scope.password) {
			alert("两次输入的密码不一致！")
			return;
		}else {
			userService.add($scope.entity,$scope.code).success(function (response) {
				if (response.success){
					alert(response.message);
				} else {
					alert(response.message);
				}
			})
		}
	};
	/*
		2.定义点击发送短信验证码，验证码进行发送的方法
	 */
	$scope.sendSmsCode=function () {
		//先进行前台正则校验，校验失败，直接打断方法弹出错误信息
		var reg_telephone =
			new RegExp("^(13[0|9]|14[5|7]|15[0|1|2|3|5|6|7|8|9]|18[0|1|2|3|5|6|7|8|9])\\d{8}$");
		if ($scope.entity.phone == null || $scope.entity.phone == "") {
			alert("请输入手机号！")
			return;
		}else if (!reg_telephone.test($scope.entity.phone)){
			alert("手机号码不合法！");
			return;
		} else{
			//调用userService中发送验证码的方法，将验证码进行发送
			userService.sendSmsCode($scope.entity.phone).success(function (response) {
				alert(response.message);
			})
		}
	}
});	

File: pinyougou-user-web/src/main/java/com/pinyougou/user/controller/UserController.java
package com.pinyougou.user.controller;
import java.util.List;

import com.pinyougou.userService.UserService;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.web.bind.annotation.RequestBody;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RestController;
import com.alibaba.dubbo.config.annotation.Reference;
import com.pinyougou.pojo.TbUser;
import com.pinyougou.userService.UserService;

import entity.PageResult;
import entity.Result;
import util.PhoneFormatCheckUtils;

/**
 * controller
 * @author Administrator
 *
 */
@RestController
@RequestMapping("/user")
public class UserController {

	@Reference
	private UserService userService;

	/**
	 * 返回全部列表
	 * @return
	 */
	@RequestMapping("/findAll")
	public List<TbUser> findAll(){			
		return userService.findAll();
	}

	/**
	 * 返回全部列表
	 * @return
	 */
	@RequestMapping("/findPage")
	public PageResult  findPage(int page,int rows){			
		return userService.findPage(page, rows);
	}
	
	/**
	 * 增加
	 * @param user
	 * @return
	 */
	@RequestMapping("/add")
	public Result add(@RequestBody TbUser user,String code){
		if(!userService.checkSmsCode(user.getPhone(),code)){
			return new Result(false,"验证码输入错误");
		}
		try {
			userService.add(user);
			return new Result(true, "注册成功！");
		} catch (Exception e) {
			e.printStackTrace();
			return new Result(false, "注册失败！");
		}
	}
	
	/**
	 * 修改
	 * @param user
	 * @return
	 */
	@RequestMapping("/update")
	public Result update(@RequestBody TbUser user){
		try {
			userService.update(user);
			return new Result(true, "修改成功");
		} catch (Exception e) {
			e.printStackTrace();
			return new Result(false, "修改失败");
		}
	}	
	
	/**
	 * 获取实体
	 * @param id
	 * @return
	 */
	@RequestMapping("/findOne")
	public TbUser findOne(Long id){
		return userService.findOne(id);		
	}
	
	/**
	 * 批量删除
	 * @param ids
	 * @return
	 */
	@RequestMapping("/delete")
	public Result delete(Long [] ids){
		try {
			userService.delete(ids);
			return new Result(true, "删除成功"); 
		} catch (Exception e) {
			e.printStackTrace();
			return new Result(false, "删除失败");
		}
	}
	
		/**
	 * 查询+分页
	 * @param
	 * @param page
	 * @param rows
	 * @return
	 */
	@RequestMapping("/search")
	public PageResult search(@RequestBody TbUser user, int page, int rows  ){
		return userService.findPage(user, page, rows);		
	}

	/**
	 * 注册时发送短信验证码
	 * @param phone 注册时提交的手机号
	 * @return 发送短信验证码的结果
	 */
	@RequestMapping("/sendSmsCode.do")
	public Result sendSmsCode(String phone){
		//为保证程序的健壮性，对mobile进行非空和正则校验
		if (phone == null || "".equals(phone)){
			return new Result(false,"请输入手机号！");
		}else if (!PhoneFormatCheckUtils.isChinaPhoneLegal(phone)){
			return new Result(false,"请输入正确格式的手机号！");
		}else{
			try {
				userService.sendSmsCode(phone);
				return new Result(true,"短信发送成功！");
			} catch (Exception e) {
				e.printStackTrace();
				return new Result(true,"短信发送失败！");
			}
		}
	}
}
